### PR TITLE
feat: Use user_id query for exporting my schedule ICS

### PR DESCRIPTION
**Pull request by Tlazypanda**
For https://github.com/fossasia/open-event-frontend/issues/6694

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@iamareebjamal there are some other places dependent on export function so should I keep user_id as default value there? :sweat_smile:

File: app/api/helpers/calendar/ical.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-from flask_jwt_extended import current_user
 from icalendar import Calendar, Event
 from sqlalchemy import or_
 from sqlalchemy.orm import joinedload
 
 from app.models.session import Session
 
 
-def to_ical(event, include_sessions=False, my_schedule=False):
+def to_ical(event, user_id, include_sessions=False, my_schedule=False):
```
Comment (iamareebjamal) [inline app/api/helpers/calendar/ical.py:9]: Should be optional, like other params. This is not a required param to export my schedule

File: app/api/helpers/calendar/ical.py [hunk 2]
```diff
@@ -39,10 +37,8 @@ def to_ical(event, include_sessions=False, my_schedule=False):
             .order_by(Session.starts_at.asc())
         )
         if my_schedule:
-            if not current_user:
```
Comment (iamareebjamal) [inline app/api/helpers/calendar/ical.py:42]: It should also work with current user

File: app/api/helpers/ICalExporter.py [hunk 3]
```diff
@@ -4,9 +4,9 @@
 
 class ICalExporter:
     @staticmethod
-    def export(event_id):
+    def export(event_id, user_id):
```
Comment (iamareebjamal) [inline app/api/helpers/ICalExporter.py:7]: No change needed in this file

File: app/api/custom/calendars.py [hunk 4]
```diff
@@ -14,13 +13,15 @@
     '/<string:event_identifier>.ics',
 )
 @to_event_id
-@jwt_optional
```
Comment (iamareebjamal) [inline app/api/custom/calendars.py:17]: No need to remove this

File: tests/all/integration/api/helpers/test_icalexporter.py [hunk 5]
```diff
@@ -5,15 +5,15 @@
 from tests.factories.session import SessionSubFactory
 
 
-def test_export_basic(db):
+def test_export_basic(db, user):
```
Comment (iamareebjamal) [inline tests/all/integration/api/helpers/test_icalexporter.py:8]: No change needed in this file

File: app/api/helpers/calendar/ical.py (hunk 2 above)
Comment (Tlazypanda) [reply app/api/helpers/calendar/ical.py:42]: So we are also keeping the jwt_optional header then too right? :sweat_smile:

**Comment by Tlazypanda** [comment]
@iamareebjamal ohh okay got it I thought we are removing header completely :sweat_smile:

File: app/api/helpers/calendar/ical.py (hunk 2 above)
Comment (iamareebjamal) [reply app/api/helpers/calendar/ical.py:42]: Yes

File: app/api/helpers/calendar/ical.py [hunk 6]
```diff
@@ -8,7 +7,7 @@
 from app.models.session import Session
 
 
-def to_ical(event, include_sessions=False, my_schedule=False):
+def to_ical(event, include_sessions=False, my_schedule=False, user_id=current_user.id):
```
Comment (iamareebjamal) [inline app/api/helpers/calendar/ical.py:10]: Default should be none. Python default arguments are evaluated once on load time, so this will never work

Comment (Tlazypanda) [reply app/api/helpers/calendar/ical.py]: Yes just pushing that change only :sweat_smile: realised that after pushing :p

File: app/api/helpers/calendar/ical.py [hunk 7]
```diff
@@ -40,8 +40,8 @@ def to_ical(event, include_sessions=False, my_schedule=False, user_id=current_us
         if my_schedule:
             if not (current_user or user_id):
                 return jsonify(error='Forbidden Access'), 401
-            if not user_id:
-                user_id = current_user.id
+
+            user_id = current_user.id if not user_id else user_id
```
Comment (iamareebjamal) [inline app/api/helpers/calendar/ical.py:45]: user_id overrides current_user

Comment (Tlazypanda) [reply app/api/helpers/calendar/ical.py:45]: Sir I have set it to current_user.id if user_id is not set :sweat_smile: so user_id will override current_user

File: app/api/helpers/calendar/ical.py [hunk 8]
```diff
@@ -39,10 +39,12 @@ def to_ical(event, include_sessions=False, my_schedule=False):
             .order_by(Session.starts_at.asc())
         )
         if my_schedule:
-            if not current_user:
-                return jsonify(error='Login Required'), 401
+            if not (current_user or user_id):
+                return jsonify(error='Forbidden Access'), 401
+
+            user_id = current_user.id if not user_id else user_id
```
Comment (iamareebjamal) [inline app/api/helpers/calendar/ical.py:45]:
```suggestion
            user_id = user_id or current_user.id
```

File: app/api/helpers/calendar/ical.py [hunk 9]
```diff
@@ -39,10 +39,12 @@ def to_ical(event, include_sessions=False, my_schedule=False):
             .order_by(Session.starts_at.asc())
         )
         if my_schedule:
-            if not current_user:
-                return jsonify(error='Login Required'), 401
+            if not (current_user or user_id):
+                return jsonify(error='Forbidden Access'), 401
```
Comment (iamareebjamal) [inline app/api/helpers/calendar/ical.py:43]:
```suggestion
                return jsonify(error='Login or User ID required'), 401
```

File: app/api/helpers/calendar/ical.py (hunk 8 above)
Comment (Tlazypanda) [reply app/api/helpers/calendar/ical.py]: Ohh I need to start writing better code :joy:

**Pull request merged**